.github/workflows/workflow.yml

```diff
@@ -226,11 +227,77 @@ jobs:
         run: |
           cd target/${{ matrix.target }}/release
           strip starship
+          echo "${{ matrix.target }} $(openssl dgst -sha256 starship)" >> ../../../${{ matrix.target }}-shasum
           tar czvf ../../../${{ matrix.name }} starship
           cd -
 
+      - name: Upload build artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.name }}
+          path: ${{ matrix.name }}
+
+      - name: Upload shasum artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.target }}-shasum
+          path: ${{ matrix.target }}-shasum
+
+  # Create GitHub release with Rust build targets and release notes
+  github_release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    name: Create GitHub Release
+    needs: github_build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Download shasums from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: x86_64-unknown-linux-gnu-shasum
+          path: .
+      - name: Download shasums from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: x86_64-unknown-linux-musl-shasum
+          path: .
+      - name: Download shasums from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: x86_64-apple-darwin-shasum
+          path: .
+      - name: Download shasums from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: x86_64-pc-windows-msvc-shasum
+          path: .
+      - name: Download releases from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-unknown-linux-gnu.tar.gz
+          path: .
+      - name: Download releases from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-unknown-linux-musl.tar.gz
+          path: .
+      - name: Download releases from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-apple-darwin.tar.gz
+          path: .
+      - name: Download releases from github_build
+        uses: actions/download-artifact@v1
+        with:
+          name: starship-x86_64-pc-windows-msvc.zip
+          path: .
```

Comment:
I really would have liked to not have this many actions for these files, but I couldn't find out how to get download-artifact to handle a list of artifacts.

See: https://github.com/actions/download-artifact/issues/6